src/golioth_openthread.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(golioth_openthread);
+
+#include <openthread/nat64.h>
+#include <openthread/dns_client.h>
+#include <zephyr/net/openthread.h>
+#include <openthread/error.h>
+
+static otDnsQueryConfig dns_query;
+static char *golioth_ipv6_addr;
+
+K_SEM_DEFINE(ot_dns_resolve, 0, 1);
+
+/* Callback for NAT64 IPv6 translated Golioth System Server address from the DNS query response */
+static void ot_dns_callback(otError aError, const otDnsAddressResponse *aResponse, void *aContext)
+{
+    otIp6Address golioth_addr;
```

Comment:
```
struct ot_dns_resolve_context *context = aContext;
```

and instead of `golioth_ipv6_addr` you can use `context->ipv6`. Instead of `k_sem_give(&ot_dns_resolve)` use `k_sem_give(&context->sem)`.